peps/pep-0760.rst

```diff
@@ -0,0 +1,162 @@
+PEP: 760
+Title: No More Bare Excepts
+Author: Pablo Galindo <pablogsal@python.org>, Brett Cannon <brett@python.org>
+PEP-Delegate: TBD
+Status: Draft
+Type: Standards Track
+Created: 30-Sep-2024
+Python-Version: 3.14
+
+
+Abstract
+========
+
+This PEP proposes disallowing bare ``except:`` clauses in Python's
+exception-handling syntax. Currently, Python allows catching all exceptions
+with a bare ``except:`` clause, which can lead to overly broad exception
+handling and mask important errors. This PEP suggests requiring explicit
+exception types in all except clauses, promoting more precise and intentional
+error handling.
+
+
+Motivation
+==========
+
+The current syntax allows for catching all exceptions with a bare ``except:`` clause is:
+
+.. code-block:: python
+
+   try:
+       risky_operation()
+   except:
+       handle_any_error()
+
+While this syntax can be convenient for a "catch all" handler, it often leads
+to poor coding practices:
+
+1. It can mask important errors that should be propagated.
+2. It makes debugging more difficult by catching and potentially hiding
+   unexpected exceptions.
+3. It goes against the Python principle of explicit over implicit.
+
+Various linters [1]_ [2]_ [3]_ and style guides (including :pep:`8`) [4]_ [5]_
+[6]_ [7]_ discourage bare ``except`` clauses.
+
+By requiring explicit exception types, we can encourage more thoughtful and
+precise error handling:
+
+.. code-block:: python
+
+   try:
+       risky_operation()
+   except Exception as e:
+       handle_expected_error(e)
+
+
+Rationale
+=========
+
+The decision to disallow bare except clauses is based on the following
+considerations:
+
+1. Requiring specific exception types makes the programmer's intentions clear
+   and encourages thinking about what exceptions might occur.
+
+2. Catching only specific exceptions makes identifying and debugging unexpected
+   errors easier.
+
+3. Preventing overly broad exception handling reduces the risk of silently
+   ignoring critical errors.
+
+4. Many style guides and linters already discourage the use of bare except
+   clauses.
+
+
+Specification
+=============
+
+The syntax for the except clause will be modified to require an exception type.
+The grammar will be updated to remove the possibility of adding an empty
+expression in except clauses.
+
+This change disallows the bare 'except:' syntax. All except clauses must
+specify at least one exception type:
+
+.. code-block:: python
+
+try:
+    ...
+except ValueError:
+    ...
+except (TypeError, RuntimeError):
+    ...
+except Exception:
+    ...  # Still allowed, but catches all exceptions explicitly
+
+The semantics of exception handling remain unchanged, except that it will no
+longer be possible to catch all exceptions without explicitly specifying
+'Exception' or a similarly broad exception type.
```

Comment:
```suggestion
``Exception`` or a similarly broad exception type.
```

Reply:
`except Exception` would not catch all exceptions, so I think this should actually be

```suggestion
``BaseException`` or a similarly broad exception type.
```